learningequality/studio
# Handle side-loading node data from separate APIs

## Summary
### Description of the change(s) you made

- Updates the Import search to side-load nodes from either the normal API endpoint or the public API which is cached
- Removes some returned fields from search API, since it's returning only a subset of what is needed
- Adds Kolibri public API frontend utilities
- Fixes inconsistent issue with `DropdownWrapper.vue` handling string values for numeric prop

### Manual verification steps performed
1. Create a public channel, Channel A
2. Add some content to Channel A, with all the extras like thumbnail and metadata
3. Publish the channel
4. Sign out and back in
5. Do NOT open Channel A
6. Create a new channel, Channel B
7. Add new content to Channel B by using Import search
8. Verify the core details are represented by the search card, preview, and you're able to complete the import

### Screenshots (if applicable)

![Screenshot from 2023-06-13 14-15-03](https://github.com/learningequality/studio/assets/819838/4a39f034-8343-4a68-aa95-a2dadef7ef0c)


### Does this introduce any tech-debt items?

___
## Reviewer guidance
### How can a reviewer test these changes?

- Does this work for public channel searches?
- Does this work for searching your own channels?


### Are there any risky areas that deserve extra testing?



## References

Resolves: https://github.com/learningequality/studio/issues/4104
Resolves: https://github.com/learningequality/studio/issues/4102
Partially addresses: https://github.com/learningequality/studio/issues/4101

## Comments


----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
